File: app/src/main/java/com/anokmik/tripassistant/user/profile/ProfileContract.java
package com.anokmik.tripassistant.user.profile;

import com.anokmik.tripassistant.util.contract.presenter.PresenterEditable;
import com.anokmik.tripassistant.util.contract.presenter.PresenterValidator;
import com.anokmik.tripassistant.util.contract.view.ViewBack;

public interface ProfileContract {

    interface View extends ViewBack {

    }

    interface Presenter extends PresenterValidator, PresenterEditable{

        void signOut();

    }

}

File: app/src/main/kotlin/com/anokmik/tripassistant/user/profile/ProfileContract.kt
package com.anokmik.tripassistant.user.profile

import com.anokmik.tripassistant.util.contract.presenter.PresenterEditable
import com.anokmik.tripassistant.util.contract.presenter.PresenterValidator
import com.anokmik.tripassistant.util.contract.view.ViewBack

interface ProfileContract {

    interface View : ViewBack

    interface Presenter : PresenterValidator, PresenterEditable {

        fun signOut()

    }

}